src/main/java/io/stargate/sgv2/jsonapi/api/model/command/Filterable.java

```diff
@@ -1,10 +1,16 @@
 package io.stargate.sgv2.jsonapi.api.model.command;
 
 import io.stargate.sgv2.jsonapi.api.model.command.clause.filter.FilterClause;
+import io.stargate.sgv2.jsonapi.api.model.command.clause.filter.FilterSpec;
 
 /*
  * All the commands that needs FilterClause will have to implement this.
  */
 public interface Filterable {
-  FilterClause filterClause();
+  FilterSpec filterSpec();
+
+  default FilterClause filterClause(CommandContext<?> ctx) {
```

Comment:
Smallest changes to existing non-test-code is to add this default method here. Also takes care of issue of `null` (missing or explicit null) for `filter` in JSON.